src/ui/public/courier/data_source/_abstract.js

```diff
@@ -1,14 +1,17 @@
 import _ from 'lodash';
 import angular from 'angular';
-import CourierRequestQueueProvider from 'ui/courier/_request_queue';
-import CourierErrorHandlersProvider from 'ui/courier/_error_handlers';
-import CourierFetchFetchProvider from 'ui/courier/fetch/fetch';
-import CourierDataSourceDecorateQueryProvider from 'ui/courier/data_source/_decorate_query';
+
+import 'ui/promises';
```

Comment:
Oh wait, I think I understand now.  This is likely only working by accident because of the way angular handles modules, but if you were to somehow try to run this service by itself, it wouldn't have the promise services loaded that are necessary.  So adding this back in is basically a noop right now, but it is technically more appropriate than not having it.